app/src/main/java/ch/epfl/skysync/database/FirestoreDatabase.kt

```diff
@@ -62,6 +62,40 @@ class FirestoreDatabase(private val useEmulator: Boolean = false) {
         }
   }
 
+  /**
+   * Set a new item to the database
+   *
+   * Set item at id (if null a new id is generated) and override any previously set id.
+   *
+   * @param path A filesystem-like path that specify the location of the table
+   * @param item The item to set to the database, the types of the attributes have to be Firestore
+   * @param id the id of the item (if null a new id is generated)
+   * @param onCompletion Callback called on completion of the operation
+   * @param onError Callback called when an error occurs
+   */
+  fun <T : Any> setItem(
+      path: String,
+      id: String?,
```

Comment:
In the method `setItem`, you expect to receive an `id`, so you should mark the `id` argument as non-nullable (that way you don't find yourself calling `additem`)

Reply:
OK, thanks. I was wondering if the id could be null, and if so, I didn't see the difference with add.